_staging/src/k8s.io/client-go/copy.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# PREREQUISITES: run `godep restore` in the main repo before calling this script.
+MAIN_REPO_FROM_SRC=${1:-"k8s.io/kubernetes"}
+MAIN_REPO=${GOPATH%:*}/src/${MAIN_REPO_FROM_SRC}
+CLIENT_REPO_FROM_SRC=${2:-"k8s.io/client-go"}
+CLIENT_REPO=${MAIN_REPO}/_staging/src/${CLIENT_REPO_FROM_SRC}
+CLIENT_REPO_TEMP=${CLIENT_REPO}/_tmp
+
+# working in the ${CLIENT_REPO_TEMP} so 'godep save' won't complain about dirty working tree.
+echo "creating the _tmp directory"
+mkdir -p ${CLIENT_REPO_TEMP}
+
+# mkcp copies file from the main repo to the client repo, it creates the directory if it doesn't exist in the client repo.
+function mkcp() {
+    mkdir -p "${CLIENT_REPO_TEMP}/$2" && cp -r "${MAIN_REPO}/$1" "${CLIENT_REPO_TEMP}/$2"
+}
+
+echo "copying client packages"
+mkcp "pkg/client/clientset_generated/release_1_4" "pkg/client/clientset_generated"
+mkcp "/pkg/client/record/" "/pkg/client"
+mkcp "/pkg/client/cache/" "/pkg/client"
+# TODO: make this test file not depending on pkg/client/unversioned
+rm ${CLIENT_REPO_TEMP}/pkg/client/cache/listwatch_test.go
+mkcp "/pkg/client/restclient" "/pkg/client"
+mkcp "/pkg/client/testing" "/pkg/client"
+# remove this test because it imports the internal clientset
+rm ${CLIENT_REPO_TEMP}/pkg/client/testing/core/fake_test.go
+mkcp "/pkg/client/transport" "/pkg/client"
+mkcp "/pkg/client/typed" "/pkg/client"
+
+mkcp "/pkg/client/unversioned/auth" "/pkg/client/unversioned"
+mkcp "/pkg/client/unversioned/clientcmd" "/pkg/client/unversioned"
+mkcp "/pkg/client/unversioned/portforward" "/pkg/client/unversioned"
+# remove this test because it imports the internal clientset
+rm ${CLIENT_REPO_TEMP}/pkg/client/unversioned/portforward/portforward_test.go
+
+pushd ${CLIENT_REPO_TEMP}
+echo "generating vendor/"
+GO15VENDOREXPERIMENT=1 godep save ./...
+popd
+
+echo "move to the client repo"
+# clean the ${CLIENT_REPO}
+ls ${CLIENT_REPO} | grep -v 'copy.sh\|_tmp' | xargs rm -r
```

Comment:
Ah. Missed the -v option. Sorry.

If you're curious, the only way to accomplish the negative matching with just `rm` is to `shopt -s extglob`. Then you could `rm -r !(copy.sh|_tmp)`